protocol/driver.go

```diff
@@ -65,12 +67,19 @@ func DialOptions() []grpc.DialOption {
 	return opts
 }
 
-func RegisterDriver(srv *grpc.Server, d driver.Driver) {
-	RegisterDriverServer(srv, &driverServer{d: d})
+// RegisterDriver registers a v2 driver server on a given gRPC server.
+func RegisterDriver(srv *grpc.Server, d driver.Host) {
+	s := &driverServer{d: d}
+	RegisterDriverServer(srv, s)
+	RegisterDriverHostServer(srv, s)
 }
 
-func AsDriver(cc *grpc.ClientConn) driver.Driver {
-	return &client{c: NewDriverClient(cc)}
+// AsDriver creates a v2 driver client for a given gRPC client.
+func AsDriver(cc *grpc.ClientConn) driver.Host {
```

Comment:
It seems odd that `AsDriver` returns a `Host`. Which I guess "is" a Driver, kinda, but the naming seems odd.

Reply:
Will be fixed when merge interfaces.